velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,246 @@
+    });
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    exec::Aggregate::setAllNulls(groups, indices);
+    for (auto i : indices) {
+      *BaseAggregate ::value<int64_t>(groups[i]) = 0;
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    // Partial and final aggregations are the same.
+    extractValues(groups, numGroups, result);
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::template updateGroups<true, int64_t>(
+        groups,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          if (result < value) {
+            result = value;
+          }
+        },
+        mayPushdown);
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::updateOneGroup(
+        group,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          result = std::max(result, value);
+        },
+        [](int64_t& result, int64_t value, int /* unused */) {
+          result = value;
+        },
+        mayPushdown,
+        (int64_t)0);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup(group, rows, args[0]);
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void updateOneAccumulator(
+      char* const group,
+      std::vector<vector_size_t>& rowSizes,
+      vector_size_t idx) {
+    if (decoded_.isNullAt(idx)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set max(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = std::max(current, (int64_t)rowSizes[idx]);
+  }
+
+  void
+  doUpdate(char** groups, const SelectivityVector& rows, const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+
+    if (decoded_.isConstantMapping() && decoded_.isNullAt(0)) {
+      // There's nothing to do; all values are null.
+      return;
+    }
+
+    if (decoded_.isConstantMapping()) {
+      estimateSerializedSizes(arg, rows, 1);
+      rows.applyToSelected([&](vector_size_t i) {
+        updateOneAccumulator(groups[i], elementSizes_, 0);
+      });
+    } else {
+      estimateSerializedSizes(arg, rows, rows.countSelected());
+      rows.applyToSelected([&](vector_size_t i) {
+        updateOneAccumulator(
+            groups[i], elementSizes_, selectToElementSizesIndexMap_[i]);
```

Comment:
It might be possible to remove selectToElementSizesIndexMap_ by adding a counter here like so:

```
vector_size_t sizeIndex = 0;
rows.applyToSelected([&](vector_size_t i) {
        updateOneAccumulator(
            groups[i], elementSizes_, sizeIndex++);
}
```